modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java

```diff
@@ -0,0 +1,244 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.notification.internal.upgrade.v3_10_4;
+
+import com.liferay.document.library.kernel.model.DLFolderConstants;
+import com.liferay.notification.constants.NotificationPortletKeys;
+import com.liferay.notification.model.NotificationQueueEntry;
+import com.liferay.notification.model.NotificationTemplate;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.Repository;
+import com.liferay.portal.kernel.model.ResourceConstants;
+import com.liferay.portal.kernel.portletfilerepository.PortletFileRepository;
+import com.liferay.portal.kernel.repository.model.Folder;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.PortalUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * @author Paulo Albuquerque
+ */
+public class DeleteStaleNotificationUpgradeProcess extends UpgradeProcess {
```

Comment:
I searched in the portal and found that DeleteStaleSomething is used in the plural, like `deleteStaleDocuments()`.
Can you rename it to `DeleteStaleNotificationQueueEntriesAndNotificationTemplatesUpgradeProcess`?